node/subsystem/src/lib.rs

```diff
@@ -195,6 +197,9 @@ pub trait SubsystemContext: Send + 'static {
 /// [`Overseer`]: struct.Overseer.html
 /// [`Subsystem`]: trait.Subsystem.html
 pub trait Subsystem<C: SubsystemContext> {
+	/// Subsystem-specific prometheus metrics.
+	type Metrics: metrics::Metrics;
```

Comment:
:+1: for making `Metrics` a core component of `Subsystem`s. I think this is a very clean approach.

In the future we should probably have the discussion whether our way of recording metrics should trickle down further than the level of `Subsystem` or whether the core logic should stay generic. E.g. `sc-network` handles all metric recording within `NetworkWorker`. Lower level components bubble up the data needed to feed these metrics. On the other hand `finality-grandpa` passes down a metrics registry all the way to bottom components.